core/src/main/java/com/linecorp/armeria/server/file/FileServiceBuilder.java

```diff
@@ -57,6 +58,7 @@
     boolean canSetEntryCacheSpec = true;
     @Nullable
     HttpHeadersBuilder headers;
+    MimeTypeFunction mimeTypeFunction = defaultMimeTypeFunction();
```

Comment:
I think we should apply this at last so that
`EXTENSION_TO_MEDIA_TYPE` and `URLConnection.guessContentTypeFromName(path)` are eventually called when the `MimeTypeFunction` specified by a user returns `null`.
So, how about making this nullable and adding the default function in `build()` method?
```java
@Nullable
MimeTypeFunction mimeTypeFunction;

public FileService build() {
    ...
    final MimeTypeFunction mimeTypeFunction;
    if (this.mimeTypeFunction == null) {
        mimeTypeFunction = MimeTypeFunction.ofDefault();
    } else {
        mimeTypeFunction = this.mimeTypeFunction.orElse(MimeTypeFunction.ofDefault());
    }
    ...
}
```